docs/demography.md

```diff
@@ -1514,16 +1514,109 @@ modelled in the oldest epochs.
 
 (sec_demography_importing)=
 
-## Importing model definitions
+## Importing/exporting model definitions
 
 (sec_demography_importing_demes)=
 
 ### Demes
 
-```{eval-rst}
-.. todo:: Details of how we import Demes models.
+The [Demes format](https://popsim-consortium.github.io/demes-spec-docs/)
+is intended to become a standard for writing demographic models,
+to facilitate interchange between simulators, as well as between
+demographic inference software and simulators.
+Demes models are typically written as a YAML file (see the
+[Demes tutorial](demes-spec:sec_tutorial) for details), and loaded into Python
+using the [demes](demes:sec_introduction) library.
+The {meth}`.Demography.from_demes` method converts a {class}`demes.Graph`
+object into a {class}`.Demography` object, and the {meth}`.Demography.to_demes`
+method does the opposite conversion.
+
+```{warning}
+Demes uses similar terminology to msprime, but uses different conventions.
+In particular, Demes uses forwards-time conventions.
+```
+
+Demes models have time in units of "time ago", and when using generations as
+the time units, the times will correspond to msprime's notion of time:
+time 0 is "now" and time values increase towards the past.
+However, in contrast to msprime, Demes uses forwards-time conventions
+everywhere else. Noteworthy differences between Demes and msprime include:
+
+* Each deme has its own list of epochs, whereas in msprime's
+  {class}`.DemographyDebugger`, epochs are time intervals that apply across
+  the entire demography.
+* Demes uses `start_time` and `end_time` to indicate time intervals,
+  but the `start_time` has a value which is larger (more ancient)
+  than the `end_time`.
+* For migrations and pulses, `source` indicates the deme in which a migrant
+  is born and the `dest` indiciates the deme in which a migrant has offspring.
+
+Consider the following Demes model that describes two extant demes `A` and `B`,
+which have a common ancestor `X`.
+
+```{literalinclude} demes_example_01.yaml
+:language: yaml
+```
+
+When visualised using [demesdraw](https://github.com/grahamgower/demesdraw),
+the demographic model looks like this:
```

Comment:
This isn't rendering for me - it's just blank. It looks like it's mucking up the other stuff below as well - maybe ditch the SVG fanciness if not absolutely necessary?

Reply:
That's strange, it looks fine to me in chrome and firefox. Did you get an error message at all? Maybe there's some version discrepancy between our environments which is important?

Reply:
Ah, maybe my build is just broken. I've ended up in a collision of dependencies around sphinx etc which I'm not motivated to solve right now - this stuff is moving annoyingly quickly.

Reply:
I also seemed to get this issue at one point, after shuffling some code things around. `make clean; make` resolved it though, and I can't seem to reproduce the problem at will.

Reply:
I also seemed to get this issue at one point, after shuffling some code things around. `make clean; make` resolved it though, and I can't seem to reproduce the problem at will.